### PR TITLE
Deploy sequences separately other than combining with actions

### PR DESCRIPTION
**Pull request by daisy-ycguo**
Fix issue #391 

In the existing source code, creating `sequence` objects for deployment is using the same method as `action`, which combining sequences deployment with actions. This issue resulted random deployment sequence of `action` and `sequence`. It's the root cause of random issue #391.

In this PR, I created new codes to create sequence objects, other than using the same method of creating actions.

### CODE REVIEW

File: deployers/manifestreader.go [hunk 1]
```diff
@@ -220,7 +220,41 @@ func (reader *ManifestReader) checkAction(action utils.ActionRecord) error {
...
+
+	for _, seqAction := range actions {
+		// check if the sequence action is exist in actions
+		// If the sequence action exists in actions, return error
+		_, exists := reader.serviceDeployer.Deployment.Packages[seqAction.Packagename].Actions[seqAction.Action.Name]
+		if exists == true {
+			return errors.New("manifestReader. Error: Conflict sequence action with an action. "
+				+ "Found a sequence action with the same name of an action:"
+				+seqAction.Action.Name)
+		}
```
Comment (pritidesai) [inline deployers/manifestreader.go]: @daisy-ycguo this is resulting in syntax error, please change it to:

```
			return errors.New("manifestReader. Error: Conflict sequence action with an action. " +
				"Found a sequence action with the same name of an action:" +
				seqAction.Action.Name)
```

**Comment by daisy-ycguo** [comment]
@pritidesai Thank you for reviewing.
I understsand sequence is a kind of action by design. 

But, method `ManifestReader.SetActions` will add actions to `reader.serviceDeployer.Deployment.Packages.Action`, and method `ManifestReader.SetSequences` will invoke  `SetActions`, which will add actions (which is sequence instead) to `reader.serviceDeployer.Deployment.Packages.Action`. So you will see `process-change-cloudant-sequence` is treated same as `process-change` ( refer to below output of wskdeploy ). All `actions` and `sequences` are saved in a list, which will cause a random deploy sequence of `actions` and `sequences`, and will result issue #391 .
```
Packages:
Name: cloudant-trigger
    bindings: 
  * dependency: openwhisk-cloudant
    location: /whisk.system/cloudant

  * action: process-change
    bindings: 
    annotations: 
  * action: process-change-cloudant-sequence
    bindings: 
    annotations: 


Triggers:
* trigger: data-inserted-trigger
    bindings: 
        - dbname : ""
    annotations: 
        - name: feed value: openwhisk-cloudant/changes

 Rules
* rule: log-change-rule
    - trigger: data-inserted-trigger
    - action: cloudant-trigger/process-change-cloudant-sequence

Do you really want to deploy this? (y/N): 
```

My PR has updated method `ManifestReader.SetSequences` to add actions (which is sequence instead) to `reader.serviceDeployer.Deployment.Packages.Sequence`. After applying my PR, the output help message of wskdeploy will look like below:
```
Packages:
Name: cloudant-trigger
    bindings: 
  * dependency: openwhisk-cloudant
    location: /whisk.system/cloudant

  * action: process-change
    bindings: 
    annotations: 

  * sequence: process-change-cloudant-sequence

Triggers:
* trigger: data-inserted-trigger
    bindings: 
        - dbname : ""
    annotations: 
        - name: feed value: openwhisk-cloudant/changes

 Rules
* rule: log-change-rule
    - trigger: data-inserted-trigger
    - action: cloudant-trigger/process-change-cloudant-sequence
```

The deploying sequences will be executed after deploying actions. And all wsk client invocations are synchronized. Then sequences will be created after actions. The issue will be fixed.

**Comment by mrutkows** [comment]
This issue may be dependent on https://github.com/apache/incubator-openwhisk-wskdeploy/issues/394

**Comment by daisy-ycguo** [comment]
@pritidesai Please review again. If it runs OK in your laptop, please help to merge it.

**Comment by daisy-ycguo** [comment]
@pritidesai If you run it with `wskdeploy -i`, you will see the difference.

**Pull request merged**